bgpd/bgpd.c

```diff
@@ -8138,3 +8136,16 @@ void bgp_gr_apply_running_config(void)
 		gr_router_detected = false;
 	}
 }
+
+printfrr_ext_autoreg_p("BP", printfrr_bp);
+static ssize_t printfrr_bp(struct fbuf *buf, struct printfrr_eargs *ea,
+			   const void *ptr)
+{
+	const struct peer *peer = ptr;
+	char peer_buf[BUFSIZ];
+
+	snprintfrr(peer_buf, sizeof(peer_buf), "%s(%s)", peer->host,
```

Comment:
use `bprintfrr` instead of `snprintfrr` + `bputs` please, saves a bunch of moving around with the extra buffer :)

Reply:
also might be good to check for `peer == NULL` here

Reply:
Amazing, thanks!

Reply:
Fixed.